### PR TITLE
Clemente tp biblioteca

### PR DESCRIPTION
**Pull request by TinoCle**

### CODE REVIEW

**Review by gonzalogrisafi — APPROVED** [review]
all things seems to be ok, i'd only change a couple of things that doesn't matter that much

File: Clemente/db/db.manager.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  getTable("books").forEach(book => {
+    if (book.id == id && book.inventory != 0) {
+      book["inventory"] -= 1;
+    }
+  });
+  save();
+}
+
+// booleano que indica si se le venció una deuda a un partner
+let hasBedt = (id, time) => {
```
Comment (florgm) [inline Clemente/db/db.manager.js]: hasDebt maybe? Wrong spelling

**Review by emicalvacho — APPROVED** [review]
Extremadamente avanzado

File: Clemente/db/db.manager.js (hunk 1 above)
Comment (TinoCle) [reply Clemente/db/db.manager.js]: my bad XD

**Review by florgm — APPROVED** [review]
I think everything is ok now

File: Clemente/handlers/app.handler.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    }
+  },
+  deletePartner: (req, res) => {
+    let id = req.params.id;
+    if (valid.id(id)) {
+      let partner = db.getPartner(id);
+      if (partner) {
+        console.log(`DELETE /partners/${id} OK`);
+        db.deletePartner(id);
+        res.status(200).json({ message: "Deleted." });
```
Comment (valdazo) [inline Clemente/handlers/app.handler.js]: Probably a better result code for a sucessful delete is 204...
Take a look to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200

File: Clemente/handlers/app.handler.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        res.status(200).json(book);
+      } else {
+        res.status(404).json({ message: "The book doesn't exist." });
+      }
+    } else {
+      console.log(`GET /books/${id} INVALID ID`);
+      res.status(400).json({ message: "Invalid id." });
+    }
+  },
+  addBook: (req, res) => {
```
Comment (valdazo) [inline Clemente/handlers/app.handler.js]: How do we change the number of copies of a book. We can add one by one, how can we decrease it.

File: Clemente/handlers/app.handler.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  },
+  deleteBook: (req, res) => {
+    let id = req.params.id;
+    let book = db.getBook(id);
+
+    if (book != null) {
+      if (book.inventory != 0) {
+        console.log(`DELETE /books/${id} OK`);
+        res.status(200).json({ message: "Deleted" });
+        db.deleteBook(id);
```
Comment (valdazo) [inline Clemente/handlers/app.handler.js]: Before deleting a book, we need to know if there are still loans of that book

File: Clemente/handlers/app.handler.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  listAllLoans: (req, res) => {
+    console.log(`GET /loans/`);
+    res.status(200).json(db.getTable("loans"));
+  },
+  lentBook: (req, res) => {
+    let { Bid } = req.body;
+    let { Pid } = req.body;
+    let book = db.getBook(Bid);
+    let partner = db.getPartner(Pid);
+    if (valid.id(Pid) && valid.id(Bid) && valid.string(book) && valid.string(partner)) {
```
Comment (valdazo) [inline Clemente/handlers/app.handler.js]: We could have many loans of the same book for the same partner, how can we identify a proper loan given that each loan has it loan date.

**Review by valdazo — CHANGES_REQUESTED** [review]

File: Clemente/handlers/app.handler.js (hunk 2 above)
Comment (TinoCle) [reply Clemente/handlers/app.handler.js]: You're right, thanks!

File: Clemente/handlers/app.handler.js (hunk 3 above)
Comment (TinoCle) [reply Clemente/handlers/app.handler.js]: In the new version: If you add a book that is already on the inventory, it just increases the number of copies.
If you delete a book, it decreases the number of copies, if it reaches 0, the book is completely deleted from the database.

File: Clemente/handlers/app.handler.js (hunk 4 above)
Comment (TinoCle) [reply Clemente/handlers/app.handler.js]: Solved in the new version.

File: Clemente/handlers/app.handler.js (hunk 5 above)
Comment (TinoCle) [reply Clemente/handlers/app.handler.js]: To simplify things, a partner can lent each book once.

**Review by valdazo — APPROVED** [review]

**Pull request merged**